### DISCUSSION TITLE
Send message (text and voice) as user

### DISCUSSION THREAD

**Discussion by a4smanjorg5** [discordjs/discord.js#7534] [Q&A] [answered]
How to send message as user (not bot)?
Please tell me the steps and give the coding.

**Comment by almostSouji** [comment, marked as answer]
> 16 You may not use self-bots or user-bots to access Discord.
<https://discord.com/guidelines>

**Reply by a4smanjorg5** [reply]
Is there a way to use OAuth2?

**Reply by a4smanjorg5** [reply]
I think what you describe is if you login without going through OAuth2

**Reply by almostSouji** [reply]
There is no OAuth scope that lets you send messages as the user: <https://discord.com/developers/docs/topics/oauth2#shared-resources-oauth2-scopes>

**Comment by asrvd** [comment]
Sending message as user is possible using self bots but **it is against discord's TOS**, so your account could get banned for doing it. Still if you want to do it, there are plenty of tutorials about this on the web.
